src/server-daemon.ts

```diff
@@ -1,31 +1,40 @@
-import * as fs from 'fs';
 import * as net from "net";
-import * as path from "path";
 import * as vscode from 'vscode';
+import * as path from "path";
 import * as child_process from "child_process";
 import * as features from './features';
 
+
 import { LanguageClientOptions, RevealOutputChannelOn } from "vscode-languageclient";
 import { LanguageClient, ServerOptions, StreamInfo } from "vscode-languageclient/node";
+import { findJavaExecutable } from "./find";
 
 type Progress = vscode.Progress<{ message?: string; increment?: number }>;
 
-export async function startDaemon(context: vscode.ExtensionContext, lspExec: string, progress: Progress) {
+export async function startDaemon(context: vscode.ExtensionContext, lspLoadPath: string, progress: Progress) {
   progress.report({ message: "Starting Aya", increment: 500 });
   const config = vscode.workspace.getConfiguration("aya");
 
   const outputChannel = vscode.window.createOutputChannel("Aya");
   context.subscriptions.push(outputChannel);
-  outputChannel.appendLine(`Aya Language Server: ${lspExec}`);
+  outputChannel.appendLine(`Aya Language Server: ${lspLoadPath}`);
 
   let mode: string = config.get<string>("lsp.mode") ?? "client";
   let port: number = config.get<number>("lsp.port") ?? 11451;
   let host: string = config.get<string>("lsp.host") ?? "localhost";
 
   let serverOptions: ServerOptions;
-  if (mode === "server") serverOptions = runServer(outputChannel, lspExec, host, port);
-  else if (mode === "client") serverOptions = runClient(host, port);
-  else serverOptions = runDebug(outputChannel, lspExec);
+  let extname = path.extname(lspLoadPath);
+  if (extname === ".jar") {
```

Comment:
Does that mean I can put the fatJar path to `aya.lsp.exec`? Then what's the point of having two different configurations?

Reply:
Yeah. I've replaced `aya.lsp.exec` and `aya.lsp.fatJar` with `aya.lsp.path`.

Now users can comment one of `"aya.lsp.path"` in `settings.json` to decide where the Aya Language Server is loaded from.